src/dfcx_scrapi/tools/search_util.py

```diff
@@ -497,6 +533,360 @@ def find_list_parameters(self, agent_id):
 
         return params_list
 
+    def get_param_presets_df(self, flow_name_list=None):
+        """Gets a dataframe of parameter presets.
+
+        Args:
+          flow_name_list (optoinal): list of names of flows to search.
+          By default all flows will be searched.
+
+        Returns:
+          Dataframe of parameter presets with columns:
+            flow,
+            page,
+            intent,
+            event,
+            route_group,
+            condition,
+            form_filling,
+            location,
+            param_name,
+            param_value
+        """
+
+        flow_id_list = []
+        if flow_name_list is None:
+            flow_id_list = list(self.flows_map.values())
+        elif isinstance(flow_name_list, list):
+            for flow_name in flow_name_list:
+                flow_id_list.append(self.flows_map[flow_name])
+        elif isinstance(flow_name_list, str):
+            if flow_name_list in self.flows_map:
+                flow_id_list = list(self.flows_map[flow_name_list])
+            else:
+                raise TypeError("Flow not found")
+        else:
+            raise TypeError(
+                "flow_name_list parameter needs to be a list of flow names"
+            )
+
+        param_dfs = []
+        for flow_id in flow_id_list:
+            #Start Page
+            start_page = self.flow_data[flow_id]
+            flow_name = start_page.display_name
+            df = self.process_param_presets_in_page(flow_name, start_page)
+            param_dfs.append(df)
+            #All Other Pages
+            for page in self.page_data[flow_id].values():
+                #page_name = page.display_name
+                df = self.process_param_presets_in_page(flow_name, page)
+                param_dfs.append(df)
+        #Combine Lists to DataFrame
+        if len(param_dfs) > 0:
+            return pd.concat(param_dfs).reset_index(drop=True)
+        return None
+
+    def process_param_presets_in_page(self, flow_name, page):
```

Comment:
@SeanScripts @skyyeat it looks like this should be a private internal function utilized by `get_param_presets_df`.

Update the semantics of the name to be `_process_param_presets_in_page()` and then move the method up above the instance methods so it's not confused as something that the user should try to call first hand.

Reminder that the style for method sorting is as follows for this library:

- Class
- Static
- Private
- Instance

Ex:
```py
@classmethod
def my_class_method(cls):
   ....

@staticmethod
def my_static_method():
   ...

def _some_private_method(self):
   ...

def my_instance_method(self):
   self._some_private_method()
   self.my_static_method()
```

This provides a nice layout for each class so that users can understand which methods they should interact with first hand and which methods are used internally by other methods.